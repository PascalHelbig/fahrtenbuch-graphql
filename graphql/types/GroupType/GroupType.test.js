const { makeExecutableSchema, addMockFunctionsToSchema } = require('graphql-tools');
const { graphql } = require('graphql');
const GroupType = require('./GroupType');

it('has to be a valid schema', () => {
  const SchemaDefinition = 'schema { query: Group }';
  const schema = makeExecutableSchema({ typeDefs: [SchemaDefinition, GroupType] });
  addMockFunctionsToSchema({
    schema,
    mocks: {
      Group: () => ({
        id: () => '1337',
        name: () => 'Club Name',
        is_club: () => true,
        members: () => [{ id: 1338 }, { id: 1339 }],
        boats: () => [{ id: 1340 }, { id: 1341 }],
      }),
    },
  });

  const query = `
    query {
      id
      name
      is_club
      members { id }
      boats { id }
    }
  `;

  return graphql(schema, query).then(result =>
    expect(result).toMatchSnapshot(),
  );
});
