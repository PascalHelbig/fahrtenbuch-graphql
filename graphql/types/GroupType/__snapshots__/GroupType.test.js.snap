exports[`test has to be a valid schema 1`] = `
Object {
  "data": Object {
    "boats": Array [
      Object {
        "id": "1340",
      },
      Object {
        "id": "1341",
      },
    ],
    "id": "1337",
    "is_club": true,
    "members": Array [
      Object {
        "id": "1338",
      },
      Object {
        "id": "1339",
      },
    ],
    "name": "Club Name",
  },
}
`;
